### PR TITLE
Adding CyberGhost VPN

### PR DESCRIPTION
**Pull request by vlaquest**
This config was for France. I'm not sure how to get more servers...

### CODE REVIEW

**Comment by vlaquest** [comment]
when getting the config files from this VPN provider we get 4 files.
 - The ovpn file that I have renamed. As explained in the wiki I have changed the paths in order to make them absolute.
 - 2 crt files
 - 1 key file

File: openvpn/cyberghost/default.ovpn [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ping-exit 60
+ping-timer-rem
+explicit-exit-notify 2
+script-security 2
+remote-cert-tls server
+route-delay 5
+verb 4
+
+
+ca /etc/openvpn/custom/ca.crt
```
Comment (pkishino) [inline openvpn/cyberghost/default.ovpn:26]: the crt and key files should be put under a cyberghost folder,
create a new folder in openvpn git folder and then set the path here to:
/etc/openvpn/cyberghost/ca.crt etc..

File: openvpn/cyberghost/default.ovpn [hunk 2]
```diff
@@ -0,0 +1,31 @@
+client
+remote 87-1-fr.cg-dialup.net 443
+dev tun 
+proto udp
+auth-user-pass
```
[the remark this answers is not in the document]
Comment (vlaquest) [reply openvpn/cyberghost/default.ovpn:5]: Yes thanks that was my mistake. I've repaired that.

File: openvpn/cyberghost/default.ovpn (hunk 1 above)
Comment (vlaquest) [reply openvpn/cyberghost/default.ovpn:26]: Yes I'll do that

Comment (pkishino) [reply openvpn/cyberghost/default.ovpn:26]: Are you planning to update the PR with the fixes so it can be merged??

Comment (vlaquest) [reply openvpn/cyberghost/default.ovpn:26]: Yes but not before 3 weeks because I'm not at home at the moment...

**Comment by ilike2burnthing** [comment]
@vlaquest ping

**Comment by vlaquest** [comment]
I intend to finish the work, sorry for the long pause. I'll get back to it soon.

**Comment by argaume** [comment]
any news on this?

**Comment by vlaquest** [comment]
@pkishino Hi
1/ I'm just wondering, how to handle the personal client.crt and client.key files ?
I know it is not valid to use the "cyberghost" folder to put the files (since it is "inside" the container)
I've also seen the possibility to include the text blocks of those files inside the ovpn file... but again where to put it.
What's the best way ?
(sorry since I've been using custom from the beginning, I don't know well regular providers)

2/ the ovpn file I gave is for France/Paris, I won't easily get other configs other countries/cities... But it should be easy to integrate new ones. Should we duplicate the file and give two names like this ?
 - default.ovpn
 - France-Paris.ovpn
Because Cyberghost looks like it organized by Country/City

**Comment by argaume** [comment]
@vlaquest I would really like if you could write a short guide on how to create a custom vpn based on cyberghost data if that's possible (and for a noob like) and how to include it to the container?

**Comment by ilike2burnthing** [comment]
https://haugene.github.io/docker-transmission-openvpn/supported-providers/#using_a_custom_provider

**Comment by vlaquest** [comment]
@pkishino Hi, I still need guidance on this, willing to help, but I have questions.

**Comment by krswin** [comment]
is it already working with cyberghost? can some one merge thanks.

**Comment by pkishino** [comment]
The link above explains how to use custom provider file, it’s quite straight forward, no plans to merge in any providers moving forward as the plan is to separate them completely

**Comment by vlaquest** [comment]
@pkishino Hi Patrick, so the plan is to no longer merge new providers ? You provide integrated support only for those currently in the product ? But the rest will have to use custom providers ?
Whatever your choice, using custom providers is not easy for everyone. I fought a few days before making it work, reading carefully for your guide.
If you don't want to integrate more support for existing providers, I think practical guidance / examples could be provided, that would "complete" your current guide.

**Comment by ilike2burnthing** [comment]
As I (not a project maintainer here) understand it, the plan going forward is to move all current (and add future) provider configs to a separate repo, hence why no new ones are being added now, as it just adds work later.

The configs will then be fetched from that other repo when needed, rather than already being in the container, nearly all just wasting space. It will also reduce the frequency of updates to this repo, as a lot (most?) of them are just updating provider configs.

Between then and now though, the custom provider option can be used.

If there's something specific missing from the guide which helped you, I'm sure you can make suggested amendments (probably best to do that in a separate issue though).

**Comment by pkishino** [comment]
as @ilike2burnthing mentioned.. we want to REMOVE ALL providers from this repo and split it up as providers keep changing their configs etc and it is a hassle to maintain this. 
If you think some more examples or such are needed, feel free to suggest one and we can review it before adding it to the documentation

**Comment by pkishino** [comment]
closing for now as not completed

**Pull request closed without merging**